Modules/Daq/src/DaqTask.cxx

```diff
@@ -36,63 +36,55 @@ using namespace o2::header;
-
-    name = system.second + "/RdhSizes";
-    title = "RDH sizes for " + system.second + ";bytes";
-    mSubSystemsRdhSizes[system.first] = new TH1F(name.c_str(), title.c_str(), 128, 0, 2047);
-    mSubSystemsRdhSizes[system.first]->SetCanExtend(TH1::kXaxis);
-  }
+  mInputRecordPayloadSize = std::make_unique<TH1F>("inputRecordSize", "Total payload size per InputRecord;bytes",
+                                                   getIntParam("inputRecordSizeBins", 128),
+                                                   getIntParam("inputRecordSizeMin", 0),
+                                                   getIntParam("inputRecordSizeMax", 2047));
+  getObjectsManager()->startPublishing(mInputRecordPayloadSize.get(), PublicationPolicy::Forever);
+
+  mInputSize = std::make_unique<TH1F>("payloadSizeInputs", "Payload size of the inputs;bytes",
+                                      getIntParam("payloadSizeInputsBins", 128),
+                                      getIntParam("payloadSizeInputsMin", 0),
+                                      getIntParam("payloadSizeInputsMax", 2047));
+  getObjectsManager()->startPublishing(mInputSize.get(), PublicationPolicy::Forever);
+
+  mNumberRDHs = std::make_unique<TH1F>("numberRdhs", "Number of RDHs per InputRecord",
+                                       getIntParam("numberRDHsBins", 100),
+                                       getIntParam("numberRDHsMin", 1),
+                                       getIntParam("numberRDHsMax", 100));
+  getObjectsManager()->startPublishing(mNumberRDHs.get(), PublicationPolicy::Forever);
+
+  mSumRDHSizesPerInputRecord = std::make_unique<TH1F>("sumRdhSizesPerInputRecord", "Sum of RDH sizes per InputRecord;bytes",
+                                                      getIntParam("sumRdhSizesPerInputRecordBins", 128),
+                                                      getIntParam("sumRdhSizesPerInputRecordMin", 0),
+                                                      getIntParam("sumRdhSizesPerInputRecordMax", 2047));
+  getObjectsManager()->startPublishing(mSumRDHSizesPerInputRecord.get(), PublicationPolicy::Forever);
+
+  mSumRDHSizesPerRDH = std::make_unique<TH1F>("RdhSizes", "RDH sizes; bytes",
+                                              getIntParam("RdhSizesBins", 128),
+                                              getIntParam("RdhSizesMin", 0),
+                                              getIntParam("RdhSizesMax", 2047));
+  getObjectsManager()->startPublishing(mSumRDHSizesPerRDH.get(), PublicationPolicy::Forever);
+
+  mRDHSizesPerCRUIds = std::make_unique<TH2F>("RdhPayloadSizePerCRUid", "RDH payload size per CRU",
+                                              getIntParam("RdhPayloadSizeBins", 128),
+                                              getIntParam("RdhPayloadSizeMin", 0),
+                                              getIntParam("RdhPayloadSizeMax", 2047),
+                                              getIntParam("CRUidBins", 500),
```

Comment:
Yes, I had this thought just now. We can check with Pippo then.

Reply:
Pippo basically confirmed that it is [12 bits set by users](https://github.com/AliceO2Group/AliceO2/blob/dev/DataFormats/Headers/include/Headers/RAWDataHeader.h#L95), so it can be anything, as every detector team does their own thing.

Reply:
oh well. let's then leave it like this, but set the default max cru ID to `2^12-1`. If we need something better, we will get a request.